src/main/java/frc/robot/commands/DriveCommands.java

```diff
@@ -61,8 +69,14 @@ private static Translation2d getLinearVelocityFromJoysticks(double x, double y)
         .getTranslation();
   }
 
-  public static Command autoAlignToReef() {
-    return Commands.run(() -> {});
+  public static Command GoToPose(Supplier<Pose2d> targetPose) {
+    return Commands.runOnce(
+        () -> {
+          Pose2d currentTarget = targetPose.get();
+          Logger.recordOutput("Dynamic Target Pose", currentTarget);
+          Command alignCommand = AutoBuilder.pathfindToPose(currentTarget, PATH_CONSTRAINTS);
+          alignCommand.schedule(); // Schedule the dynamically generated command
```

Comment:
Should just return the alignCommand instead of having a command to generate another command

Reply:
Tried that, but the command needs to change every time it is run, and if I do it the normal way, It only ever gives the the path that is created on instantiation.

Reply:
Consider ScheduleCommand or ProxyCommand if you really need to do it this way
https://docs.wpilib.org/en/stable/docs/software/commandbased/command-compositions.html#scheduling-other-commands

Reply:
Actually DeferredCommand is probably what you want to use